openedx/features/edly/middleware.py

```diff
@@ -0,0 +1,24 @@
+"""
+Edly Organization Access Middleware
+"""
+from logging import getLogger
+from django.http import Http404
+
+from openedx.features.edly.utils import user_has_edly_organization_access
+
+logger = getLogger(__name__)
+
+
+class EdlyOrganizationAccessMiddleware(object):
+    """
+    Django middleware to validate edly user organization access based on request.
+    """
+
+    def process_request(self, request):
+        """
+        Validate logged in user's access based on request site and its linked edly sub organization.
+        """
+
+        if request.user.is_authenticated and not user_has_edly_organization_access(request):
+            logger.warning('Edly user %s has no access for site %s.' % (request.user.email, request.site))
```

Comment:
Please add test for the log message.

Reply:
Unit Test updated to check log message on 404.